main/glossary/README.md

```diff
@@ -67,151 +104,379 @@ synchronously compared for structural equivalence.
-stream of updates describing changes to the state of an offer.
-
-See more: [Notifier](/distributed-programming.md)
+A notifier provides a stream of updates describing changes to the state of an [offer](#offer).
+For more information, see the [Notifier section in the Distributed JavaScript Programming Guide](/distributed-programming.md#notifiers).
 
 ## Object Capabilities
 
-Objects have state, behavior, and references. Lets say Object A has references to Objects B and C, while B and C do not have references to each other. Thus, A can communicate with B and C, and B and C cannot communicate with each other.
-There is an effective zero-cost firewall between B and C.
+Objects have state, behavior, and references. Lets say Object A has references to Objects B 
+and C, while B and C do not have references to each other. Thus, A can communicate with B and C, 
+and B and C cannot communicate with each other. There is an effective zero-cost firewall between B and C.
 
-An *object capability system* constrains how references are obtained. You can't get one just by knowing the name of a global variable or a public class. You can get a reference in only three ways. 
+An *object capability system* constrains how references are obtained. You can't get one just by 
+knowing the name of a global variable or a public class. You can only get a reference via: 
 - Creation: Functions that create objects get a reference to them.
 - Construction: Constructors can endow their constructed objects with references, including inherited references. 
 - Introduction: 
   - A has references to B and C. 
-  - B and C  do *not* have references to each other
+  - B and C do *not* have references to each other
   - A sends B a reference to C. 
     - B now has a reference to C and can communicate with C. 
 
-If references can only be obtained by creation, construction, or introduction, you may have a safe system. If they can be obtained in any other way, your system is unsafe.
+If references can only be obtained by creation, construction, or introduction, you may have a safe 
+system. If they can be obtained in any other way, your system is unsafe.
 
 For more information, see [Douglas Crockford on Object Capabilities](https://frontendmasters.com/courses/good-parts-javascript-web/object-capabilities/).
 
+## Offer
+
+Offers are a structured way of describing user intent. In Zoe, an offer consists of a [proposal](#proposal) (the 
+rules under which the party wants to exercise the offer) and [payments](#payment) corresponding to what the proposal specifies as what the
+party will give if the offer is satisfied. The payments are automatically [escrowed](#escrow) by Zoe, and appropriately reallocated when
+the offer exits with either success or rejection. See [`E(Zoe).offer(invitation, proposal, paymentKeywordRecord)`](https://agoric.com/documentation/zoe/api/zoe.html#e-zoe-offer-invitation-proposal-paymentkeywordrecord).
```

Comment:
I feel like I've been repeating myself for months, but there's no general concept of an "offer being satisfied". We need to strike any such language post-haste! Also **no reallocation happens on offer exits!** I think I've said this a number of times. There is a fundamental misunderstanding of what reallocation means that we need to correct.

```suggestion
conditions under which the party makes the offer) and [payments](#payment) corresponding to what the party is willing to give in exchange for what they want. The payments are automatically [escrowed](#escrow) by Zoe, and reallocated according to the contract code. An offer gets a payout of some combination of what the party originally contributed and what others have contributed. The specific payout is determined by the contract code. See [`E(Zoe).offer(invitation, proposal, paymentKeywordRecord)`](https://agoric.com/documentation/zoe/api/zoe.html#e-zoe-offer-invitation-proposal-paymentkeywordrecord).
```